thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomalydetection/model/detection/MinMaxThresholdDetectionModel.java

```diff
@@ -0,0 +1,85 @@
+package com.linkedin.thirdeye.anomalydetection.model.detection;
+
+import com.linkedin.thirdeye.anomalydetection.context.AnomalyDetectionContext;
+import com.linkedin.thirdeye.anomalydetection.context.TimeSeries;
+import com.linkedin.thirdeye.api.DimensionMap;
+import com.linkedin.thirdeye.datalayer.dto.RawAnomalyResultDTO;
+import java.util.ArrayList;
+import java.util.List;
+import org.joda.time.Interval;
+
+public class MinMaxThresholdDetectionModel extends AbstractDetectionModel {
+  public static final String DEFAULT_MESSAGE_TEMPLATE = "change : %.2f %%, currentVal : %.2f, min : %.2f, max : %.2f";
+  public static final String MIN_VAL = "min";
+  public static final String MAX_VAL = "max";
+
+  @Override
+  public List<RawAnomalyResultDTO> detect(String metricName,
+      AnomalyDetectionContext anomalyDetectionContext) {
+    List<RawAnomalyResultDTO> anomalyResults = new ArrayList<>();
+
+    // Get min / max props
+    Double min = null;
+    if (properties.containsKey(MIN_VAL)) {
+      min = Double.valueOf(properties.getProperty(MIN_VAL));
+    }
+    Double max = null;
+    if (properties.containsKey(MAX_VAL)) {
+      max = Double.valueOf(properties.getProperty(MAX_VAL));
+    }
+
+    TimeSeries timeSeries = anomalyDetectionContext.getTransformedCurrent(metricName);
+
+    // Compute the weight of this time series (average across whole)
+    double averageValue = 0;
+    for (long time : timeSeries.timestampSet()) {
+      averageValue += timeSeries.get(time);
+    }
+    // Compute the bucket size, so we can iterate in those steps
+    long bucketMillis = anomalyDetectionContext.getBucketSizeInMS();
+    Interval timeSeriesInterval = timeSeries.getTimeSeriesInterval();
+    long numBuckets = Math.abs(timeSeriesInterval.getEndMillis() - timeSeriesInterval.getStartMillis()) / bucketMillis;
+
+    // avg value of this time series
+    averageValue /= numBuckets;
+
+    DimensionMap dimensionMap = anomalyDetectionContext.getTimeSeriesKey().getDimensionMap();
+    for (long timeBucket : timeSeries.timestampSet()) {
+      double value = timeSeries.get(timeBucket);
+      double deviationFromThreshold = getDeviationFromThreshold(value, min, max);
+
+      if (deviationFromThreshold != 0) {
+        RawAnomalyResultDTO anomalyResult = new RawAnomalyResultDTO();
+        anomalyResult.setProperties(properties.toString());
+        anomalyResult.setStartTime(timeBucket);
+        anomalyResult.setEndTime(timeBucket + bucketMillis); // point-in-time
+        anomalyResult.setDimensions(dimensionMap);
+        anomalyResult.setScore(averageValue);
+        anomalyResult.setWeight(deviationFromThreshold); // higher change, higher the severity
```

Comment:
I think the original design of the weight of Min Max function is that if the time series swings below and above the min max thresholds, then the weight should be countered.

I agree that the weight calculation for this function should be redesigned, which is out of the scope of this migration.